ignite/pkg/protoanalysis/protoutil/cursor.go

```diff
@@ -0,0 +1,302 @@
+package protoutil
```

Comment:
For the method in this package

Is there a reason of using `panic` instead of returning an error in the method?

Reply:
Hey! This kept the same logic as in `go/astutil`. Don't know what their rationale was but the way I see it these panics are meant for us, the developers, if it panics it means we've done something exceptionally silly while traversing an AST with no real possibility of recovery.